code/game/machinery/teleporter/console.dm

```diff
@@ -0,0 +1,187 @@
+/obj/machinery/computer/teleporter
+	name = "Teleporter Control Console"
+	desc = "Used to control a linked teleportation hub and station."
+	icon_keyboard = "teleport_key"
+	icon_screen = "teleport"
+
+	var/obj/machinery/tele_projector/projector
+	var/obj/machinery/tele_pad/pad
+	var/atom/target
+	var/active
+	var/id
+
+
+/obj/machinery/computer/teleporter/Destroy()
+	clear_target()
+	if (projector)
+		projector.lost_computer()
+	clear_projector()
+	if (pad)
+		pad.lost_computer()
+	clear_pad()
+	. = ..()
+
+
+/obj/machinery/computer/teleporter/Initialize()
+	. = ..()
+	id = "[random_id(/obj/machinery/computer/teleporter, 1000, 9999)]"
+	update_refs()
+
+
+/obj/machinery/computer/teleporter/proc/update_refs()
+	for (var/dir in GLOB.cardinal)
+		projector = locate() in get_step(src, dir)
+		if (projector)
+			break
+	if (projector)
+		var/pad_dir
+		for (var/dir in GLOB.cardinal)
+			pad = locate() in get_step(projector, dir)
+			if (pad)
+				pad_dir = dir
+				break
+		if (pad)
+			projector.set_computer(src)
+			projector.set_dir(pad_dir)
+			pad.set_computer(src)
+			projector.queue_icon_update()
+			pad.queue_icon_update()
+
+
+/obj/machinery/computer/teleporter/proc/clear_projector()
+	if (!projector)
+		return
+	GLOB.destroyed_event.unregister(projector, src, /obj/machinery/computer/teleporter/proc/lost_projector)
+	projector = null
+	set_active(FALSE)
+
+
+/obj/machinery/computer/teleporter/proc/lost_projector()
+	audible_message(SPAN_WARNING("\The [src] buzzes, \"Projector missing.\""))
+	clear_projector()
+
+
+/obj/machinery/computer/teleporter/proc/set_projector(obj/machinery/tele_projector/_projector)
```

Comment:
Is this actually called?

Reply:
Unsure why these were added but not actually used. I'll update the `update_refs()` proc to use these instead of updating the var directly.